src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -321,6 +355,32 @@ public Activity AddTag(string key, string? value)
             return this;
         }
 
+        /// <summary>
+        /// Add <see cref="ActivityLink" /> object to the <see cref="Links" /> list.
+        /// </summary>
+        /// <param name="link"> object of <see cref="ActivityLink"/> to add to the attached links list.</param>
+        /// <returns>'this' for convenient chaining</returns>
+        public Activity AddLink(ActivityLink link)
+        {
+            // Cannot set Link after starting the Activity.
+            if (StartTimeUtc != default)
```

Comment:
This could generate a false positive:

````
    Activity a = new Activity("");
    a.SetStartTime(...)
    a.AddLink(...) // shouldn't throw, but would
    a.Start();
````

The [check that Start() uses](https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Activity.cs,455) is: 
if (_id != null || _spanId != null)